--- Chain_Bug_2_works.thy	2013-09-07 16:26:03.695088000 +0200
+++ Chain_Bug_1_fails.thy	2013-09-07 16:17:11.108179000 +0200
@@ -1,4 +1,4 @@
-theory Chain_Bug_2_works
+theory Chain_Bug_1_fails
 imports Main
 begin
 
@@ -16,6 +16,8 @@
 notepad
 begin
   fix Y::"'b set"
+  have "{{}} = injections {} Y"
+  proof -
   have "{{}} = {R::('a \<times> 'b) set. Domain R = {} \<and> Range R \<subseteq> Y \<and> runiq R \<and> runiq (R\<inverse>)}" (is "?LHS = ?RHS")
   proof
     have "Domain {} = {}" by simp
@@ -38,5 +40,6 @@
   qed
   also have "{R . Domain R = {} \<and> Range R \<subseteq> Y \<and> runiq R \<and> runiq (R\<inverse>)} = injections {} Y"
     unfolding injections_def ..
-  finally have "{{}} = injections {} Y" .
+    finally show "{{}} = injections {} Y" sorry
+  qed
 end
